File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
    answer = 0
    if n % 2 == 1:
        return sum([x for x in range(n + 1) if x % 2 == 1])
    else:
        return sum([x**2 for x in range(n + 1) if x % 2 == 0])

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        
        for (int i = 1; i <= n; i++) {
            
            if (i % 2 == 0 && n % 2 == 0) {
                answer += Math.pow(i, 2);
            } else if (i % 2 == 1 && n % 2 == 1) {
                answer += i;
            }
        }
        
        return answer;
    }
}